invokeai/app/services/shared/sqlite_migrator/migrations/migration_4.py

```diff
@@ -0,0 +1,94 @@
+import sqlite3
+
+from invokeai.app.services.shared.sqlite_migrator.sqlite_migrator_common import Migration
+
+
+class Migration4Callback:
+    """Callback to do step 4 of migration."""
+
+    def __call__(self, cursor: sqlite3.Cursor) -> None:  # noqa D102
+        self._create_model_metadata(cursor)
+        self._create_model_tags(cursor)
+        self._create_tags(cursor)
+        self._create_triggers(cursor)
+
+    def _create_model_metadata(self, cursor: sqlite3.Cursor) -> None:
+        """Create the table used to store model metadata downloaded from remote sources."""
+        cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS model_metadata (
+                id TEXT NOT NULL,
+                name TEXT GENERATED ALWAYS AS (json_extract(metadata, '$.name')) VIRTUAL NOT NULL,
+                author TEXT GENERATED ALWAYS AS (json_extract(metadata, '$.author')) VIRTUAL NOT NULL,
+                -- Serialized JSON representation of the whole metadata object,
+                -- which will contain additional fields from subclasses
+                metadata TEXT NOT NULL,
+                created_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                -- Updated via trigger
+                updated_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+                FOREIGN KEY(id) REFERENCES model_config(id)
+            );
+            """
+        )
+
+    def _create_model_tags(self, cursor: sqlite3.Cursor) -> None:
+        cursor.execute(
+            """--sql
+            CREATE TABLE IF NOT EXISTS model_tags (
+                id TEXT NOT NULL,
```

Comment:
Minor: I think this would be a little clearer.
```suggestion
                model_id TEXT NOT NULL,
```

Reply:
Changed to `model_id`